src/lib.rs

```diff
@@ -44,13 +44,7 @@
 //! ```{rust}
 //! use semver::Version;
 //!
-//! assert!(Version::parse("1.2.3") == Ok(Version {
-//!    major: 1,
-//!    minor: 2,
-//!    patch: 3,
-//!    pre: vec!(),
-//!    build: vec!(),
-//! }));
+//! assert!(Version::parse("1.2.3") == Ok(Version::new(1, 2, 3)));
```

Comment:
:+1:

I think we should add an example that also sets some pre and patch values, just to see how that API looks. I guess making that ergonomic was what you were getting at with the `set_pre` and `set_build` methods?

I know it was this way before you changed it, but this could also be to an `assert_eq`.